packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -69,17 +71,33 @@ export class ResourceDownloadQueue {
     }
   }
 
+  public cancelAllRequests(): void {
+    for (const key in this._requestsToCancel) {
+      this._requestsToCancel[key].abortController.abort();
+    }
+    this._requestsToCancel = {};
```

Comment:
Why not just call the `cancelRequest` method from here?

Reply:
The thought here was that since we are doing a for loop on the `this_requestsToCancel` I didn't want to modify the the object while we were looping through it cause it would be poor CS practice, and there were ways we could circumvent this but this just seemed cleaner and easier to read IMO.